src/index.ts

```diff
@@ -2,33 +2,42 @@
 
 import ReactExports from 'react';
 
+type Usable<T> = PromiseLike<T> & {
+  status?: 'pending' | 'fulfilled' | 'rejected';
+  value?: T;
+  reason?: unknown;
+} | React.Context<T>;
+
+function isContext<T>(usable: Usable<T>): usable is React.Context<T> {
+  return '_currentValue' in usable && '$$typeof' in usable;
+}
+
 export const use =
   ReactExports.use ||
   (<T>(
-    promise: PromiseLike<T> & {
-      status?: 'pending' | 'fulfilled' | 'rejected';
-      value?: T;
-      reason?: unknown;
-    },
+    usable: Usable<T>,
   ): T => {
-    if (promise.status === 'pending') {
-      throw promise;
-    } else if (promise.status === 'fulfilled') {
-      return promise.value as T;
-    } else if (promise.status === 'rejected') {
-      throw promise.reason;
+    if (isContext(usable)) {
+      return ReactExports.useContext(usable);
```

Comment:
Does it require `// eslint-disable-next-line ...`?

Reply:
I do get a warning for using ReactExports.useContext _instead of_ having a named import for `useContext` - however it doesn't error about conditional `useContext` usage like I'd expect.

Reply:
Oh, I see it now, probably because it isn't considered as a hook.